app/src/main/java/com/konifar/annict/viewmodel/MyWorksViewModel.java

```diff
@@ -0,0 +1,147 @@
+package com.konifar.annict.viewmodel;
+
+import android.content.Context;
+import android.databinding.ObservableInt;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+
+import com.annimon.stream.Collectors;
+import com.annimon.stream.Stream;
+import com.konifar.annict.api.AnnictClient;
+import com.konifar.annict.model.Status;
+import com.konifar.annict.model.Works;
+import com.konifar.annict.pref.DefaultPrefs;
+import com.konifar.annict.util.PageNavigator;
+import com.konifar.annict.view.fragment.MyWorksFragment;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+import rx.Subscription;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
+
+public class MyWorksViewModel implements ViewModel {
+
+    private static final String TAG = MyWorksViewModel.class.getSimpleName();
+
+    private final Context context;
+    private final AnnictClient client;
+    private final PageNavigator pageNavigator;
+    private final CompositeSubscription compositeSubscription = new CompositeSubscription();
+
+    private int currentPage = 1;
+    private boolean isLoading;
+
+    private Status status = Status.NON;
+
+    public ObservableInt progressBarVisibility = new ObservableInt(View.GONE);
+    public ObservableInt recyclerViewVisibility = new ObservableInt(View.GONE);
+    public ObservableInt footerProgressBarVisibility = new ObservableInt(View.GONE);
+
+    @Inject
+    public MyWorksViewModel(Context context,
+                            AnnictClient client,
+                            PageNavigator pageNavigator) {
+        this.context = context;
+        this.client = client;
+        this.pageNavigator = pageNavigator;
+    }
+
+    public void setStatus(@NonNull Status status) {
+        this.status = status;
+    }
+
+    public void showNextWorks(MyWorksFragment.MyWorksAdapter adapter) {
```

Comment:
Hmm, it's not good to pass adapter class.
So I'll fix to call `subscribe()` method in Fragment class.